src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -47,16 +52,30 @@ interface DeepCloneOptions {
  * @param options  - Additional options which configure the diff operation
  * @returns An object of the data in other which differs from that in original.
  */
-export declare function diffObject(original: object, other: object, { inner }?: DiffObjectOptions): object;
+export declare function diffObject(original: object, other: object, options?: DiffObjectOptions): object;
 
 interface DiffObjectOptions {
   /**
    * Only recognize differences in other for keys which also exist in original
    * @defaultValue `false`
    */
   inner?: boolean;
+
+  /**
+   * Apply special logic to deletion keys. They will only be kept if the original object has a
+   * corresponding key that could be deleted.
+   * @defaultValue `false`
+   */
+  deletionKeys?: boolean;
 }
 
+/**
+ * Test if two objects contain the same enumerable keys and values.
+ * @param a - The first object.
+ * @param b - The second object.
+ */
+export function objectsEqual(a: Record<string, unknown>, b: Record<string, unknown>): boolean;
```

Comment:
It also picks up `number` keys. Also, you can pass in objects that don't have an index signature. So I would argue that we should probably just type `a` and `b` as `object`.